actors/miner/tests/declare_recoveries.rs

```diff
@@ -0,0 +1,174 @@
+use fil_actor_miner::pledge_penalty_for_continued_fault;
+use fil_actor_miner::power_for_sectors;
+use fil_actors_runtime::test_utils::expect_abort_contains_message;
+use fvm_ipld_bitfield::BitField;
+use fvm_shared::address::Address;
+use fvm_shared::bigint::BigInt;
+use fvm_shared::clock::ChainEpoch;
+use fvm_shared::consensus::{ConsensusFault, ConsensusFaultType};
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::error::ExitCode;
+
+mod util;
+use crate::util::*;
+
+// an expriration ~10 days greater than effective min expiration taking into account 30 days max
+// between pre and prove commit
+const DEFAULT_SECTOR_EXPIRATION: ChainEpoch = 220;
+
+const PERIOD_OFFSET: ChainEpoch = 100;
+const BIG_BALANCE: u128 = 1_000_000_000_000_000_000_000_000u128;
+
+#[test]
+fn recovery_happy_path() {
+    let mut h = ActorHarness::new(PERIOD_OFFSET);
+    let mut rt = h.new_runtime();
+    rt.set_balance(TokenAmount::from(BIG_BALANCE));
+    h.construct_and_verify(&mut rt);
+    let one_sector =
+        h.commit_and_prove_sectors(&mut rt, 1, DEFAULT_SECTOR_EXPIRATION as u64, vec![], true);
+
+    // advance to first proving period and submit so we'll have time to declare the fault next cycle
+    h.advance_and_submit_posts(&mut rt, &one_sector);
+
+    // Declare the sector as faulted
+    h.declare_faults(&mut rt, &one_sector);
+
+    // Declare recoveries updates state
+    let st = h.get_state(&rt);
+    let (dl_idx, p_idx) =
+        st.find_sector(&rt.policy, &rt.store, one_sector[0].sector_number).unwrap();
+    h.declare_recoveries(
+        &mut rt,
+        dl_idx,
+        p_idx,
+        BitField::try_from_bits([one_sector[0].sector_number]).unwrap(),
+        BigInt::from(0),
```

Comment:
```suggestion
        BigInt::zero(),
```

Reply:
Replaced all occurrences of `BigInt::from(0)` with `BigInt::zero()`